ffi/src/utils.rs

```diff
@@ -21,18 +21,17 @@ pub mod strings {
         NamespacedKeyword,
     };
 
-    pub fn c_char_to_string(cchar: *const c_char) -> String {
+    pub fn c_char_to_string(cchar: *const c_char) -> &'static str {
         let c_str = unsafe { CStr::from_ptr(cchar) };
-        let r_str = c_str.to_str().unwrap_or("");
-        r_str.to_string()
+        c_str.to_str().unwrap_or("")
     }
 
     pub fn string_to_c_char<T>(r_string: T) -> *mut c_char where T: Into<String> {
         CString::new(r_string.into()).unwrap().into_raw()
     }
 
-    pub fn kw_from_string(mut keyword_string: String) -> NamespacedKeyword {
-        let attr_name = keyword_string.split_off(1);
+    pub fn kw_from_string(keyword_string: &'static str) -> NamespacedKeyword {
+        let attr_name = keyword_string.trim_left_matches(":");
         let parts: Vec<&str> = attr_name.split("/").collect();
```

Comment:
This will silently accept ":one/two/three" and produced ":one/two".  `panic!` instead?